### ISSUE TITLE
Descriptor's type inference does not work as expected

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#1630]
**Describe the bug**
Return type of descriptor is not inferred correctly if using TypeVar

**To Reproduce**

```
from typing import TypeVar, Type, Any


T = TypeVar('T')


class Factory:
    def __get__(self, obj: Any, cls: Type[T]) -> T:
        return cls()


class SomeClass:
    instance: Factory


reveal_type(SomeClass.instance)
```

pyright: `Type of "SomeClass.instance" is "Any"`
mypy: `Revealed type is 'SomeClass*'`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This change is included in pyright 1.1.122, which I just published. It will also be in the next release of pylance.
